### PR TITLE
Feat/group account selection

### PR DESCRIPTION
**Pull request by marie-j**
### Description of the Change

Before: Selection was not saved between pagination changes 
Now: Selection is kept 

MBP-140

### CODE REVIEW

File: src/exchange/group/accounts/group-accounts.controller.js [hunk 1]
```diff
@@ -28,94 +28,92 @@ angular.module('Module.exchange.controllers').controller(
...
 
-    resetSearch() {
-      this.search.value = null;
-      this.services.$scope.$broadcast('paginationServerSide.loadPage', 1, 'accountsByGroupTable');
-    }
+    updateManagersList(newManagerValue, account) {
+      const bufferedAccount = _.find(
+        this.accountsListBuffer.list.results,
+        bufferedAcc => bufferedAcc.id === account.id,
+      );
```
Comment (JeremyDec) [inline src/exchange/group/accounts/group-accounts.controller.js]: You can get rid of lodash :)
```suggestion
      const bufferedAccount = this.accountsListBuffer.list.result.find(({ id }) => id === account.id);
```

File: src/exchange/group/accounts/group-accounts.controller.js [hunk 2]
```diff
@@ -28,94 +28,92 @@ angular.module('Module.exchange.controllers').controller(
...
     }
 
-    saveSelection() {
-      this.model.managersList = [];
-      this.model.membersList = [];
+    updateMembersList(newMemberValue, account) {
+      const bufferedAccount = _.find(
+        this.accountsListBuffer.list.results,
+        bufferedAcc => bufferedAcc.id === account.id,
+      );
```
Comment (JeremyDec) [inline src/exchange/group/accounts/group-accounts.controller.js]: Same than above

File: src/exchange/group/accounts/group-accounts.controller.js [hunk 3]
```diff
@@ -28,94 +28,92 @@ angular.module('Module.exchange.controllers').controller(
...
-          }
+    applySelection(account) {
+      const accountInManagerList = _.find(
+        this.model.managersList,
+        manager => manager.id === account.id,
+      );
+      const accountInMemberList = _.find(
+        this.model.membersList,
+        manager => manager.id === account.id,
+      );
```
Comment (JeremyDec) [inline src/exchange/group/accounts/group-accounts.controller.js]: Same for the two than above

File: src/exchange/group/accounts/group-accounts.html [hunk 4]
```diff
@@ -1,115 +1,38 @@
...
             </div>
-        </div>
-    </div>
+        </oui-message>
+
+        <oui-datagrid data-rows-loader="$ctrl.getAccounts($config)" data-page-size="10" data-row-loader="$ctrl.applySelection($row)">
+            <oui-column data-title=":: 'exchange_GROUPS_delegation_step1_email_header' | translate" data-type="string" data-searchable>
+                <p data-ng-if="$row.message">
+                    <span class="oui-icon oui-icon-warning_circle mr-2" aria-hidden="true"></span>
+                    <span data-ng-bind="$row.id"></span>
```
Comment (JeremyDec) [inline src/exchange/group/accounts/group-accounts.html]: `::` possible no ?

File: src/exchange/group/accounts/group-accounts.html [hunk 5]
```diff
@@ -1,115 +1,38 @@
...
+               data-primary-action="$ctrl.updateAccounts()"
+               data-primary-label="{{:: 'exchange_common_confirm' | translate }}"
+               data-secondary-action="resetAction()"
+               data-secondary-label="{{:: 'exchange_common_cancel' | translate }}"
+               data-on-dismiss="resetAction()"
+               data-loading="!$ctrl.accountsList">
+
+        <p data-translate="exchange_GROUPS_accounts_step1_intro"></p>
+        <oui-message data-type="warning" data-ng-if="$ctrl.accountsList.list.messages.length > 0">
+            <div>
```
Comment (FredericEspiau) [inline src/exchange/group/accounts/group-accounts.html]:
```suggestion
```

File: src/exchange/group/accounts/group-accounts.html [hunk 6]
```diff
@@ -1,115 +1,38 @@
...
+                <span data-translate="exchange_tab_ACCOUNTS_partial"></span>
             </div>
-        </div>
-    </div>
+        </oui-message>
+
+        <oui-datagrid data-rows-loader="$ctrl.getAccounts($config)" data-page-size="10" data-row-loader="$ctrl.applySelection($row)">
+            <oui-column data-title=":: 'exchange_GROUPS_delegation_step1_email_header' | translate" data-type="string" data-searchable>
+                <p data-ng-if="$row.message">
+                    <span class="oui-icon oui-icon-warning_circle mr-2" aria-hidden="true"></span>
```
Comment (FredericEspiau) [inline src/exchange/group/accounts/group-accounts.html]: How do a screen reader know it is a warning then ?

**Pull request merged**